tests/store/model_registry/test_sqlalchemy_store.py

```diff
@@ -125,30 +125,83 @@ def test_delete_registered_model(self):
             self.store.delete_registered_model(name=name)
         assert exception_context.exception.error_code == ErrorCode.Name(RESOURCE_DOES_NOT_EXIST)
 
+    def _list_registered_models(self, page_token=None, max_results=10):
+        result = self.store.list_registered_models(page_token, max_results)
+        for idx in range(len(result)):
+            result[idx] = result[idx].name
+        return result
+
     def test_list_registered_model(self):
         self._rm_maker("A")
-        registered_models = self.store.list_registered_models()
+        registered_models = self.store.list_registered_models(page_token=None, max_results=10)
         self.assertEqual(len(registered_models), 1)
         self.assertEqual(registered_models[0].name, "A")
         self.assertIsInstance(registered_models[0], RegisteredModel)
 
         self._rm_maker("B")
-        self.assertEqual(set([rm.name for rm in self.store.list_registered_models()]),
+        self.assertEqual(set(self._list_registered_models()),
                          set(["A", "B"]))
 
         self._rm_maker("BB")
         self._rm_maker("BA")
         self._rm_maker("AB")
         self._rm_maker("BBC")
-        self.assertEqual(set([rm.name for rm in self.store.list_registered_models()]),
+        self.assertEqual(set(self._list_registered_models()),
                          set(["A", "B", "BB", "BA", "AB", "BBC"]))
 
         # list should not return deleted models
         self.store.delete_registered_model(name="BA")
         self.store.delete_registered_model(name="B")
-        self.assertEqual(set([rm.name for rm in self.store.list_registered_models()]),
+        self.assertEqual(set(self._list_registered_models()),
                          set(["A", "BB", "AB", "BBC"]))
 
+    def test_list_registered_model_paginated(self):
+        rms = [self._rm_maker(f"RM{i:03}").name for i in range(50)]
+
+        # test flow with fixed max_results
+        returned_rms = []
+        result = self._list_registered_models(page_token=None, max_results=5)
+        returned_rms.extend(result)
+        while result.token:
+            result = self._list_registered_models(page_token=result.token, max_results=5)
+            returned_rms.extend(result)
+        self.assertEqual(set(rms), set(returned_rms))
```

Comment:
this seems largely redundant with the checks below, except that here we are testing a case where the last call will return `max_results` registered models. so can we
1. make max_results maybe 25, so it doesn't loop so much and 
2. verify after the loop that `result.token` is `None` and `result` has `max_results` elements? we can also check that we loop the expected number of times.

Reply:
Added this test at Mani's suggestion earlier - updated the test so that it has a higher max_results and verified None token and that the result is of the right size